### PR TITLE
DROID-2797 Date as an object | Vault default settings

### PR DESCRIPTION
**Pull request by konstantiniiv**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: core-models/src/main/java/com/anytypeio/anytype/core_models/Constants.kt [hunk 1]
```diff
@@ -1,4 +1,19 @@
...
 
 const val NO_VALUE = ""
-const val EMPTY_QUERY = ""
\ No newline at end of file
+const val EMPTY_QUERY = ""
+
+/**
+ * The default date format pattern to use if retrieval fails.
+ */
+const val DEFAULT_DATE_PATTERN = "dd/MM/yyyy"
```
Comment (uburoiubu) [inline core-models/src/main/java/com/anytypeio/anytype/core_models/Constants.kt:9]: FALLBACK_DATE_PATTERN

File: presentation/src/test/java/com/anytypeio/anytype/presentation/widgets/collection/DateProviderImplTest.kt [hunk 2]
```diff
@@ -1,13 +1,27 @@
...
 
 class DateProviderImplTest {
 
-    val dateProviderImpl = DateProviderImpl(ZoneId.systemDefault())
+    @Mock
+    lateinit var localeProvider: LocaleProvider
+
+    init {
+        MockitoAnnotations.openMocks(this)
+        Mockito.`when`(localeProvider.locale()).thenReturn(Locale.getDefault())
```
Comment (uburoiubu) [inline presentation/src/test/java/com/anytypeio/anytype/presentation/widgets/collection/DateProviderImplTest.kt:21]: let's move into @Before test method

File: presentation/build.gradle [hunk 3]
```diff
@@ -46,6 +46,7 @@ dependencies {
     implementation libs.urlcleaner
 
     implementation libs.amplitude
+    testImplementation project(':device')
```
Comment (uburoiubu) [inline presentation/build.gradle:49]: might not be required

Comment (uburoiubu) [reply presentation/build.gradle:49]: or move your test into device module?

**Review by uburoiubu — APPROVED** [review]

File: core-models/src/main/java/com/anytypeio/anytype/core_models/Constants.kt (hunk 1 above)
Comment (konstantiniiv) [reply core-models/src/main/java/com/anytypeio/anytype/core_models/Constants.kt:9]: DONE

File: presentation/src/test/java/com/anytypeio/anytype/presentation/widgets/collection/DateProviderImplTest.kt (hunk 2 above)
Comment (konstantiniiv) [reply presentation/src/test/java/com/anytypeio/anytype/presentation/widgets/collection/DateProviderImplTest.kt:26]: done

File: presentation/build.gradle (hunk 3 above)
Comment (konstantiniiv) [reply presentation/build.gradle]: fixed

**Pull request merged**